### PR TITLE
[RISCV][GlobalISel] Select G_FRAME_INDEX

### PR DESCRIPTION
**Pull request by nitinjohnraj**
This patch is a bandage to get G_FRAME_INDEX working. We could import the SelectionDAG patterns for the ComplexPattern FrameAddrRegImm, and perhaps we will do that in the future. For now we just select it as an addition with 0.

### CODE REVIEW

**Review by mshockwave — APPROVED** [review]
LGTM. Please consider pushing new commit instead of forced push (unless it's to rebase on critical fixes) though.

File: llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp [hunk 1]
```diff
@@ -239,6 +239,19 @@ bool RISCVInstructionSelector::select(MachineInstr &MI) {
   }
   case TargetOpcode::G_SEXT_INREG:
     return selectSExtInreg(MI, MIB);
+  case TargetOpcode::G_FRAME_INDEX: {
+    // TODO: We may want to replace this code with the SelectionDAG patterns,
+    // which fail to get imported because it uses FrameAddrRegImm, which is a
+    // ComplexPattern
+    Register DstReg = MI.getOperand(0).getReg();
+
+    if (!MRI.getType(DstReg).isPointer())
```
Comment (topperc) [inline llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp:248]: Do we need this check? AArch64 has it but they were primarily checking for address space 0. The destination should always be some sort of pointer right?

Comment (nitinjohnraj) [reply llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp:248]: You're right, let me remove this

**Review by topperc — CHANGES_REQUESTED** [review]
Missing test for register bank selection

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**